src/Sylius/Component/Product/Model/Product.php

```diff
@@ -280,10 +280,10 @@ public function hasAttribute(BaseAttributeValueInterface $attribute)
     /**
      * {@inheritdoc}
      */
-    public function hasAttributeByName($attributeName)
+    public function hasAttributeByCode($attributeCode)
     {
         foreach ($this->attributes as $attribute) {
-            if ($attribute->getName() === $attributeName) {
+            if ($attribute->getAttribute()->getCode() === $attributeCode) {
```

Comment:
This `$attribute` should be renamed to `$attributeValue` because it looks weird :)

Btw.
Not sure why `$this->attributes` isn't `$this->attributesValues`...